### PR TITLE
Fixes #237: search bar added for registering repository in CI

### LINKED ISSUE #237: Add serach bar for registering reposiotory in CI
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

Now select bar is limited to choose repository from 100 repository
### Expected behaviour

make a search bar to choose from users public repository

### Steps to reproduce
1.
2.
3.

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

I have addded search bar for searching user reposiory to register in CI
## Related Issue

close #237 
## Screenshots (if appropriate):
search_bar

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by sch00lb0y** [comment]
demo deployment. https://demo08.herokuapp.com/

**Comment by pri22296** [comment]
When no result found for a search item, instead of displaying a empty dropdown, display a relevant message.

**Comment by sch00lb0y** [comment]
@pri22296 I have added,Thanks

**Comment by imujjwal96** [comment]
It is fetching some data on empty values

**Comment by imujjwal96** [comment]
It is possible to click on register button when repositories are not even searched.

**Comment by sch00lb0y** [comment]
@imujjwal96 I have fixed. please check

**Comment by imujjwal96** [comment]
@sch00lb0y When I press enter after typing something, I'd hope that it searches for that repository. Instead, it performs the action that would've been performed by clicking register.

**Comment by sch00lb0y** [comment]
@imujjwal96 that is deafult form behavior.Now, it is overridden. it'll work as you expected. Thanks for finding the issue :smile:

**Comment by imujjwal96** [comment]
@sch00lb0y The register button isn't working in your deployment

File: views/index.jade [hunk 1]
```diff
@@ -117,16 +117,73 @@ block content
...
     script.
       $("#ciModal").modal('show');
+      $('#ci_register').on('keyup keypress', function(e) {
+        var keyCode = e.keyCode || e.which;
+        if (keyCode === 13) { 
+          e.preventDefault();
+          return false;
+        }
+      });
+      let search = () => {
```
Comment (imujjwal96) [inline views/index.jade]: don't use this! Use basic function definition of JS (Structure for JS in the project should be same.)
```
var search = function () {
  ....
  ....
}
```

Also, can you try and move all this JS to 
`public/scripts/ci.js`

File: views/index.jade [hunk 2]
```diff
@@ -117,16 +117,73 @@ block content
...
+        $("#search_result").empty();
+        if ($("#search_bar").val() === "") {
+          $("#search_result").append(`<center><p>Please enter the repository name<p></center>`);
```
Comment (imujjwal96) [inline views/index.jade]: the `<center>` tag is deprecated now. Look for an alternative

Comment (sch00lb0y) [inline views/index.jade]: I have made changes

File: views/index.jade (hunk 1 above)
Comment (sch00lb0y) [inline views/index.jade]: I have made changes please check

File: public/scripts/ci.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+function search () {
+  $("#search_result").empty();
+  if ($("#search_bar").val() === "") {
+    $("#search_result").append('<p class="text-center">Please enter the repository name<p>');
+  } else {
+    $("#search_result").append('<p class="text-center">Fetching data<p>');
+    $.get(`https://api.github.com/search/repositories?q=user:${username}+fork:true+${$("#search_bar").val()}`,
+    (result) => {
+      let addDisable = function () {
```
Comment (imujjwal96) [inline public/scripts/ci.js]: can we not define this function outside?

File: public/scripts/ci.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+function search () {
+  $("#search_result").empty();
+  if ($("#search_bar").val() === "") {
+    $("#search_result").append('<p class="text-center">Please enter the repository name<p>');
+  } else {
+    $("#search_result").append('<p class="text-center">Fetching data<p>');
+    $.get(`https://api.github.com/search/repositories?q=user:${username}+fork:true+${$("#search_bar").val()}`,
+    (result) => {
```
Comment (imujjwal96) [inline public/scripts/ci.js]: here also
```
$.get(url, function (result) {
  ....
  ....
});
```

File: public/scripts/ci.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+  });
+  $('#ci_register').on('keyup keypress', function(e) {
+    var keyCode = e.keyCode || e.which;
+    if (keyCode === 13) {
+      e.preventDefault();
+      return false;
+    }
+  });
+});
```
Comment (imujjwal96) [inline public/scripts/ci.js]: add the missing semicolon at places.

**Comment by imujjwal96** [comment]
@sch00lb0y  The register button is still not working!

File: public/scripts/ci.js (hunk 3 above)
Comment (sch00lb0y) [inline public/scripts/ci.js]: I didn't find the usage of this function outside the scope. That's why i scoped inside

Comment (imujjwal96) [inline public/scripts/ci.js]: Doesn't look pretty though :)

**Comment by sch00lb0y** [comment]
@imujjwal96 I forgot to update the deployment. please check now

**Comment by imujjwal96** [comment]
I registered my app in your deployment and make commits it my repo. Nothing happened

**Comment by sch00lb0y** [comment]
I tested with my repo it is working fine. https://github.com/sch00lb0y/StockiumAI/tree/gh-pages I don't understand why it is not working. check whether webhook is registered or not. check once again with making one more commit

**Comment by imujjwal96** [comment]
Made another commit. Still didn't work

**Comment by sch00lb0y** [comment]
please raise an issue. I'll inspect why it is not working. but this PR is not making any changes with webhook

File: public/scripts/ci.js (hunk 5 above)
Comment (sch00lb0y) [inline public/scripts/ci.js]: I have added all the place. I need to add linter config on my local system to avoid this :sweat:

**Comment by mariobehling** [comment]
@pri22296 Let's try to give two reviews as quickly as possible

**Comment by pri22296** [comment]
@mariobehling I'm sorry. I forgot about this. would not happen in future

**Pull request merged**